src/solvers/dgmulti/dg.jl

```diff
@@ -34,20 +34,63 @@ mul_by_accum!(A, α) = @inline (out, x)->matmul!(out, A, x, α, one(eltype(out))
 @inline each_face_node_global(mesh::AbstractMeshData, dg::DGMulti, cache) = Base.OneTo(dg.basis.Nfq * mesh.md.num_elements)
 
 # interface with semidiscretization_hyperbolic
-wrap_array(u_ode::StructArray, mesh::AbstractMeshData, equations, dg::DGMulti, cache) = u_ode
+wrap_array(u_ode, mesh::AbstractMeshData, equations, dg::DGMulti, cache) = u_ode
 function digest_boundary_conditions(boundary_conditions::NamedTuple{Keys,ValueTypes}, mesh::AbstractMeshData,
                                     dg::DGMulti, cache) where {Keys,ValueTypes<:NTuple{N,Any}} where {N}
   return boundary_conditions
 end
 
-function allocate_coefficients(mesh::AbstractMeshData, equations, dg::DGMulti, cache)
+# Allocate nested array type for DGMulti solution storage.
+function allocate_nested_array(uEltype, nvars, array_dimensions)
+
+  # old approach: store components as separate arrays, combine via StructArrays
+  return StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(uEltype, array_dimensions...), nvars))
+
+  # TODO: any way to use a raw array of size (nvars, array_dimension...) for contiguity in memory?
+  # Can use the following:
+  #   `u_array = zeros(uEltype, nvars, array_dimensions...)`
+  #   `u = StructArray{SVector{nvars, uEltype}}(ntuple(i->view(u_array, i, ntuple(_->:, length(array_dimensions))...), nvars))`
+  # However, then `similar(u)` becomes `StructArray{<:SVector, NTuple{N, <:Matrix}...}`, and
+  # redefining `similar(u)` breaks things.
+end
+
+function create_cache(mesh::VertexMappedMesh, equations, dg::DGMultiWeakForm, RealT, uEltype)
+
+  rd = dg.basis
   md = mesh.md
+
+  # volume quadrature weights, volume interpolation matrix, mass matrix, differentiation matrices
+  @unpack wq, Vq, M, Drst = rd
+
+  # ∫f(u) * dv/dx_i = ∑_j (Vq*Drst[i])'*diagm(wq)*(rstxyzJ[i,j].*f(Vq*u))
+  weak_differentiation_matrices = map(D -> -M \ ((Vq * D)' * diagm(wq)), Drst)
+
   nvars = nvariables(equations)
-  return StructArray{SVector{nvars, real(dg)}}(ntuple(_->similar(md.x),nvars))
+
+  # storage for volume quadrature values, face quadrature values, flux values
+  u_values = allocate_nested_array(uEltype, nvars, size(md.xq))
+  u_face_values = allocate_nested_array(uEltype, nvars, size(md.xf))
+  flux_face_values = allocate_nested_array(uEltype, nvars, size(md.xf))
+  if typeof(rd.approximationType) <: SBP
+    Fscale = rd.wf ./ rd.wq[rd.Fmask] # lift scalings for diag-norm SBP operators
```

Comment:
I would prefer using a snake_case variable name that is slightly more descriptive. What about `lift_scalings` or something like that?

Reply:
Sounds good (`Fscale` is copying Warburton/Hesthaven notation, which tends to lean more terse).